sdk/core/azure-core/inc/azure/core/operation.hpp

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include "azure/core/operation_status.hpp"
+#include "azure/core/response.hpp"
+
+#include <chrono>
+
+namespace Azure { namespace Core {
+
+  //@param <T> The type of the final result of the long-running operation.
+  template <class T> class Operation {
+  protected:
+    OperationStatus m_status = OperationStatus::NotStarted;
+  public:
+    OperationStatus Status() { return m_status; }
+
+    bool Done() {
+      return (
+          m_status == OperationStatus::Succeeded ||
+          m_status == OperationStatus::Cancelled ||
+          m_status == OperationStatus::Failed);
+    }
+
+    bool HasValue()
+    {
+      return (m_status == OperationStatus::Succeeded);
+    }
+
+    virtual T Value() = 0;
```

Comment:
Removing `const` on a virtual method in the future isn't considered breaking, correct?

Reply:
Removing `const` would most definitely be breaking since it's part of the function signature; "old" implementations of `T Value() const;` in derived classes would not override the "new" `virtual T Value();` in the base class. (I should have made this more clear in my initial comment.)

Reply:
Then it is certainly a great call out. We don't want to too eagerly mark something as const if we expect it might need to be changed in the future.

What about removing const for regular non-virtual public methods? Is that fine to do?